### DISCUSSION TITLE
error while compiling esp32 port

### DISCUSSION THREAD

**Discussion by dannybr** [micropython/micropython#14154] [ESP32]
using esp-idf  idf5.3.py3.11,
did a git pull just now, but still getting:
...
In file included from /Volumes/src/micropython/micropython/py/objlist.h:29,
                 from /Volumes/src/micropython/micropython/extmod/modnetwork.c:31:
/Volumes/src/micropython/micropython/ports/esp32/modnetwork_globals.h:39:65: error: 'WIFI_AUTH_WPA3_EXT_PSK_MIXED_MODE' undeclared here (not in a function); did you mean 'MP_QSTR_AUTH_WPA3_EXT_PSK_MIXED_MODE'?
   39 | { MP_ROM_QSTR(MP_QSTR_AUTH_WPA3_EXT_PSK_MIXED_MODE), MP_ROM_INT(WIFI_AUTH_WPA3_EXT_PSK_MIXED_MODE) },
      |                                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Volumes/src/micropython/micropython/py/obj.h:90:67: note: in definition of macro 'MP_OBJ_NEW_SMALL_INT'
   90 | #define MP_OBJ_NEW_SMALL_INT(small_int) ((mp_obj_t)((((mp_uint_t)(small_int)) << 1) | 1))
      |                                                                   ^~~~~~~~~
/Volumes/src/micropython/micropython/ports/esp32/modnetwork_globals.h:39:54: note: in expansion of macro 'MP_ROM_INT'
   39 | { MP_ROM_QSTR(MP_QSTR_AUTH_WPA3_EXT_PSK_MIXED_MODE), MP_ROM_INT(WIFI_AUTH_WPA3_EXT_PSK_MIXED_MODE) },
      |                                                      ^~~~~~~~~~
[23/92] Building C object esp-idf/main_esp32/CMakeFiles/__idf_main_esp32.dir/Volumes/src/micropython/micropython/extmod/network_cyw43.c.obj
ninja: build stopped: subcommand failed.
ninja failed with exit code 1, output of the command is in the /Volumes/src/micropython/micropython/ports/esp32/build-ESP32_GENERIC/log/idf_py_stderr_output_80568 and /Volumes/src/micropython/micropython/ports/esp32/build-ESP32_GENERIC/log/idf_py_stdout_output_80568
-e See https://github.com/micropython/micropython/wiki/Build-Troubleshooting
make: *** [all] Error 1

**Comment by ned-pcs** [comment]
According to the `ports/esp32/README.md`, MicroPython supports only v5.0.4 and v5.1.2 of ESP-IDF.

**Reply by bulletmark** [reply]
Actually, the current version of `ports/esp32/README.md` says "Currently MicroPython supports v5.0.4, v5.0.5, v5.1.2, v5.2.0" but pretty clearly that does not include what the OP is using.

**Comment by dannybr** [comment]
Wups, jumped the gun again,
Sorry for the noise

**Comment by cboostwvanooijen** [comment]
Maybe useful: dockerfile for building the esp32 port (comment out line 62)
https://github.com/wovo/godafoss-code/blob/main/make/dockers/esp32/Dockerfile
